qcodes/tests/dataset/test_experiment_container.py

```diff
@@ -306,3 +313,35 @@ def test_load_last_experiment(empty_temp_db):
     assert last_exp.exp_id == exp2.exp_id
     assert last_exp.exp_id != exp1.exp_id
     assert last_exp.path_to_db == exp2.path_to_db
+
+
+@pytest.mark.usefixtures("empty_temp_db")
+def test_active_experiment():
+    NA = None
+
+    exp_id = get_active_experiment_id()
+    assert exp_id == NA
+
+    exp_1 = load_or_create_experiment("test_exp", sample_name="no_sample")
+    assert get_active_experiment_id() == exp_1.exp_id
+
+    exp_2 = new_experiment("test_exp_2", sample_name="no_sample")
+    assert get_active_experiment_id() == exp_2.exp_id
+
+    exp_3 = load_experiment(1)
+    assert get_active_experiment_id() == exp_1.exp_id
+    assert get_active_experiment_id() == exp_3.exp_id
+
+    exp_4 = new_experiment("test_exp_3", sample_name="no_sample")
+
+    exp_5 = load_experiment_by_name("test_exp_2", sample="no_sample")
+    assert get_active_experiment_id() == exp_2.exp_id
+    assert get_active_experiment_id() == exp_5.exp_id
+
+    exp_6 = load_last_experiment()
+    assert get_active_experiment_id() == exp_4.exp_id
+    assert get_active_experiment_id() == exp_6.exp_id
+
+    reset_active_experiment_id()
+    exp_id = get_active_experiment_id()
+    assert exp_id == NA
```

Comment:
```suggestion

    exp_id = get_active_experiment_id()
    assert exp_id is None

    exp_1 = load_or_create_experiment("test_exp", sample_name="no_sample")
    assert get_active_experiment_id() == exp_1.exp_id

    exp_2 = new_experiment("test_exp_2", sample_name="no_sample")
    assert get_active_experiment_id() == exp_2.exp_id

    exp_3 = load_experiment(1)
    assert get_active_experiment_id() == exp_1.exp_id
    assert get_active_experiment_id() == exp_3.exp_id

    exp_4 = new_experiment("test_exp_3", sample_name="no_sample")

    exp_5 = load_experiment_by_name("test_exp_2", sample="no_sample")
    assert get_active_experiment_id() == exp_2.exp_id
    assert get_active_experiment_id() == exp_5.exp_id

    exp_6 = load_last_experiment()
    assert get_active_experiment_id() == exp_4.exp_id
    assert get_active_experiment_id() == exp_6.exp_id

    reset_active_experiment_id()
    exp_id = get_active_experiment_id()
    assert exp_id is None
```

Reply:
I think I tried that but got error. I will check it again